scripts/library_handler.py

```diff
@@ -0,0 +1,412 @@
+Adopted from RavenUtils.py
+"""
+import os
+import sys
+import platform
+import argparse
+import subprocess
+from collections import OrderedDict
+
+from update_install_data import loadRC
+
+# python changed the import error in 3.6
+if sys.version_info[0] == 3 and sys.version_info[1] >= 6:
+  impErr = ModuleNotFoundError
+else:
+  impErr = ImportError
+
+# python 2.X uses a different capitalization for configparser
+try:
+  import configparser
+except impErr:
+  import ConfigParser as configparser
+
+try:
+  # python 3.8+ includes this in std lib
+  import importlib_metadata
+  usePackageMeta = True
+except impErr:
+  # the old way to check libs and versions uses subprocess instead
+  usePackageMeta = False
+
+# some libs are called differently in conda and within python interpreter
+## only needed for the subprocess way to check libs!
+## mapping is conda name: python import name
+libAlias = {'scikit-learn': 'sklearn',
+            'netcdf4': 'netCDF4',
+            'pyside2': 'PySide2',
+            'lazy-import': 'lazy_import',
+           }
+
+# some bad actors can't use the metadata correctly
+# and so need special treatment
+# -> see findLibAndVersion
+metaExceptions = ['pyside2', 'AMSC']
+
+skipChecks = ['python', 'hdf5', 'swig', 'nomkl']
+
+# load up the ravenrc if it's present
+## TODO we only want to do this once, but does it need to get updated?
+rcFile = os.path.abspath(os.path.join(os.path.dirname(__file__), '..', '.ravenrc'))
+if os.path.isfile(rcFile):
+  rcOptions = loadRC(rcFile)
+else:
+  rcOptions = {}
+
+#############
+#    API    #
+#############
+def inPython3():
+  """
+    Legacy method for consistency. Returns True, since we're python 3 now.
+    Leaving this here for Python 4 or similar futures.
+    @ In, None
+    @ Out, inPython3, True
+  """
+  return True
+
+def checkVersions():
+  """
+    Returns true if versions should be checked.
+    @ In, None
+    @ Out, checkVersions, bool, if true versions should be checked.
+  """
+  return os.environ.get("RAVEN_IGNORE_VERSIONS", "0") != "1"
+
+def checkLibraries(buildReport=False):
+  """
+    Looks for required libraries, and their matching QA versions.
+    @ In, buildReport, bool, optional, if True then report all libraries instead of just problems
+    @ Out, missing, list(tuple(str, str)), list of missing libraries and needed versions
+    @ Out, notQA, list(tuple(str, str, str)), mismatched versions as (libs, need version, found version)
+  """
+  missing = []
+  notQA = []
+  need = getRequiredLibs()
+  messages = []
+  for lib, needVersion in need.items():
+    # some libs aren't checked from within python
+    if lib in skipChecks:
+      continue
+    found, msg, foundVersion = checkSingleLibrary(lib, version=needVersion)
+    if not found:
+      missing.append((lib, needVersion))
+      continue
+    if needVersion is not None and foundVersion != needVersion:
+      notQA.append((lib, needVersion, foundVersion))
```

Comment:
It seems to me, the QA versions are the ones that we included in the config file, and we will not have the ranges for versions of libraries. Let's have an discussion during the stand up meeting today.